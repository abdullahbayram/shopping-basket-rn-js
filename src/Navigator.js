import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { LinearGradient } from 'expo-linear-gradient';
import { StyleSheet } from 'react-native';
import ProductListScreen from './screens/ProductListScreen/ProductListScreen';
import CheckoutScreen from './screens/CheckoutScreen/CheckoutScreen';
import ErrorScreen from './screens/ErrorScreen/ErrorScreen';
import SuccessScreen from './screens/SuccessScreen/SuccessScreen';
import PaymentScreen from './screens/PaymentScreen/PaymentScreen';

const Stack = createNativeStackNavigator();

const screenOptions = {
  headerBackground: () => (
    <LinearGradient colors={['#6fc8dc', '#8bd4c7']} style={styles.linear} start={{ x: 0, y: 0 }} end={{ x: 1, y: 1 }} />
  ),
  headerTitleStyle: {
    color: '#004f6b',
  },
  headerBackTitleVisible: false,
  headerTintColor: '#004f6b',
};

const screenOptionsResultScreens = {
  headerLeft: () => null,
  gestureEnabled: false,
  headerBackVisible: false,
};

const Navigator = () => {
  return (
    <NavigationContainer>
      <Stack.Navigator initialRouteName="ProductList">
        <Stack.Screen options={screenOptions} name="ProductList" component={ProductListScreen} />
        <Stack.Screen options={screenOptions} name="Checkout" component={CheckoutScreen} />
        <Stack.Screen options={screenOptions} name="Payment" component={PaymentScreen} />
        <Stack.Screen
          name="Error"
          options={{ ...screenOptions, ...screenOptionsResultScreens }}
          component={ErrorScreen}
        />
        <Stack.Screen
          name="Success"
          component={SuccessScreen}
          options={{ ...screenOptions, ...screenOptionsResultScreens }}
        />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default Navigator;

const styles = StyleSheet.create({
  linear: {
    flex: 1,
  },
});
