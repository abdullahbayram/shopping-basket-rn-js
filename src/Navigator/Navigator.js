import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { useTheme } from 'react-native-paper';
import PropTypes from 'prop-types';
import { ProductListScreen, CheckoutScreen, PaymentScreen, SuccessScreen, ErrorScreen } from '../screens';
import { Toggle } from '../components/molecules';
import { LinearGradient } from '../components/atoms';

const Stack = createNativeStackNavigator();

const HeaderBackground2 = () => <HeaderBackground />;
const ToggleDark = (isDarkMode, toggleTheme) => <Toggle isDarkMode={isDarkMode} toggleTheme={toggleTheme} />;

const HeaderBackground = () => {
  const { colors } = useTheme();

  return <LinearGradient colors={[colors.linearLeft, colors.linearRight]} />;
};

const screenOptionsResultScreens = {
  headerLeft: () => null,
  gestureEnabled: false,
  headerBackVisible: false,
};

const Navigator = ({ toggleTheme, isDarkMode }) => {
  const { colors } = useTheme();

  const screenOptions = () => ({
    headerBackground: () => HeaderBackground2(),
    headerTitleStyle: {
      color: colors.textPrimary,
    },
    headerBackTitleVisible: false,
    headerTintColor: colors.textPrimary,
    headerRight: () => ToggleDark(isDarkMode, toggleTheme),
  });

  return (
    <NavigationContainer testID="NavigationContainer" accessibilityRole="navigation">
      <Stack.Navigator initialRouteName="ProductList" screenOptions={screenOptions}>
        <Stack.Screen name="ProductList" component={ProductListScreen} />
        <Stack.Screen name="Checkout" component={CheckoutScreen} />
        <Stack.Screen name="Payment" component={PaymentScreen} />
        <Stack.Screen name="Error" component={ErrorScreen} options={screenOptionsResultScreens} />
        <Stack.Screen name="Success" component={SuccessScreen} options={screenOptionsResultScreens} />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

Navigator.propTypes = {
  toggleTheme: PropTypes.func.isRequired,
  isDarkMode: PropTypes.bool.isRequired,
};

export default Navigator;
