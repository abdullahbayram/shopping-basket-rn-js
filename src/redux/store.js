import { configureStore } from '@reduxjs/toolkit';
import counterReducer from './slices/counterSlice';
import { apiSlice } from './api/apiSlice';

const store = configureStore({
  reducer: {
    counter: counterReducer,
    [apiSlice.reducerPath]: apiSlice.reducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(apiSlice.middleware),
  devTools: process.env.NODE_ENV !== 'production', // Optional: Enable Redux DevTools in development mode
});

export default store;
