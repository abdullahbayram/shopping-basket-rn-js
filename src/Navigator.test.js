import React from 'react';
import { screen, waitFor } from '@testing-library/react-native';
import Navigator from './Navigator';
import renderInProvider from '../__tests__/utils/renderInProvider';
import store from './redux/store';

describe('Navigator Component', () => {
  it('should render ProductListScreen by default', async () => {
    renderInProvider(<Navigator />, { store });

    // Check that ProductListScreen is displayed by default
    await waitFor(() => {});
    expect(screen.toJSON()).toMatchSnapshot();
  });
  // Simulate navigate
  // simulate an invalid screen state
});
