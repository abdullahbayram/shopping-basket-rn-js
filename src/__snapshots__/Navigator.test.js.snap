// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigator Component should render ProductListScreen by default 1`] = `
<RNSScreenStack
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreen
    collapsable={false}
    fullScreenSwipeShadowEnabled={false}
    gestureResponseDistance={
      {
        "bottom": -1,
        "end": -1,
        "start": -1,
        "top": -1,
      }
    }
    hasLargeHeader={false}
    nativeBackButtonDismissalEnabled={false}
    onAppear={[Function]}
    onDisappear={[Function]}
    onDismissed={[Function]}
    onGestureCancel={[Function]}
    onHeaderBackButtonClicked={[Function]}
    onNativeDismissCancelled={[Function]}
    onTransitionProgress={[Function]}
    onWillAppear={[Function]}
    onWillDisappear={[Function]}
    replaceAnimation="push"
    sheetAllowedDetents="large"
    sheetCornerRadius={-1}
    sheetExpandsWhenScrolledToEdge={true}
    sheetGrabberVisible={false}
    sheetLargestUndimmedDetent="all"
    stackPresentation="push"
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    swipeDirection="horizontal"
  >
    <View
      accessibilityElementsHidden={false}
      importantForAccessibility="auto"
      style={
        {
          "flex": 1,
          "flexDirection": "column-reverse",
        }
      }
    >
      <View
        collapsable={false}
        style={
          [
            {
              "flex": 1,
            },
            {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityRole="progressbar"
          accessibilityState={
            {
              "busy": true,
            }
          }
          accessible={true}
          style={
            [
              {
                "alignItems": "center",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <View
            collapsable={false}
            style={
              {
                "height": 24,
                "opacity": 1,
                "width": 24,
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "height": 24,
                    "transform": [
                      {
                        "rotate": "45deg",
                      },
                    ],
                    "width": 24,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "height": 12,
                      "overflow": "hidden",
                      "width": 24,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "height": 24,
                        "transform": [
                          {
                            "translateY": 0,
                          },
                          {
                            "rotate": "-165deg",
                          },
                        ],
                        "width": 24,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "height": 12,
                          "overflow": "hidden",
                          "width": 24,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          {
                            "borderColor": "#c62828",
                            "borderRadius": 12,
                            "borderWidth": 2.4,
                            "height": 24,
                            "width": 24,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              collapsable={false}
              style={
                {
                  "alignItems": "center",
                  "bottom": 0,
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "height": 24,
                    "transform": [
                      {
                        "rotate": "45deg",
                      },
                    ],
                    "width": 24,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "height": 12,
                      "overflow": "hidden",
                      "top": 12,
                      "width": 24,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "height": 24,
                        "transform": [
                          {
                            "translateY": -12,
                          },
                          {
                            "rotate": "345deg",
                          },
                        ],
                        "width": 24,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "height": 12,
                          "overflow": "hidden",
                          "width": 24,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          {
                            "borderColor": "#c62828",
                            "borderRadius": 12,
                            "borderWidth": 2.4,
                            "height": 24,
                            "width": 24,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <RNSScreenStackHeaderConfig
      backButtonInCustomView={false}
      backTitleVisible={true}
      backgroundColor="rgb(255, 255, 255)"
      color="rgb(0, 122, 255)"
      direction="ltr"
      disableBackButtonMenu={false}
      hidden={false}
      hideBackButton={false}
      largeTitleHideShadow={false}
      title="Product List"
      titleColor="rgb(28, 28, 30)"
      topInsetEnabled={false}
      translucent={false}
    />
  </RNSScreen>
</RNSScreenStack>
`;
